server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -607,7 +607,8 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     }
 
     private static Map<String, DiscoveryNodeRole> rolesToMap(final Stream<DiscoveryNodeRole> roles) {
-        return Collections.unmodifiableMap(roles.collect(Collectors.toMap(DiscoveryNodeRole::roleName, Function.identity())));
+        Stream<DiscoveryNodeRole> rolesWithSearch = Stream.concat(roles, Stream.of(DiscoveryNodeRole.SEARCH_ROLE));
```

Comment:
I think we should follow up here and create a new collection like `BUILT_IN_ROLES` that is for roles that are exclusive so we can easily extend